### PR TITLE
Feature/edit text password

### PR DESCRIPTION
**Pull request by otigreros**
Add the PasswordEditTextField

### CODE REVIEW

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextPasswordField.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+class EditTextPasswordField(context: Context, attrs: AttributeSet) : BaseEditTextFormField(context, attrs) {
+
+    override fun setup() {
+        super.setup()
+
+        mEditText?.id = R.id.etPassword
+        mEditText?.transformationMethod = PasswordTransformationMethod.getInstance()
+        isPasswordVisibilityToggleEnabled = true
+        mHint = "Password"
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextPasswordField.kt]: creta a resource o constant

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        //Then
+        Assert.assertEquals(
+            ValidationResult(false, String.format(VALIDATE_EMPTY_ERROR, "Password")),
+            result
+        )
+    }
+
+    @Test
+    fun shouldShowAndErrorWithEmptyText() {
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt]: this is te same that shouldShowAndErrorWithEmptyPassword

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+        //When
+        view.perform(ViewActions.typeText("1234567890"))
+
+        //Then
+        not(isTextDisplayed("1234567890"))
+    }
+
+    @Test
+    fun tapEyeToSeePassword() {
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt]: //Given . //When .  // Then

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        shouldNotSeePassword()
+        val checkableImageButton = Espresso.onView(withId(R.id.text_input_password_toggle))
+        checkableImageButton.perform(ViewActions.click())
+        isTextDisplayed("1234567890")
+
+    }
+
+    @Test
+    fun tapEyeToHidePassword() {
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt]: //Given . //When .  // Then

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        shouldNotSeePassword()
+        val checkableImageButton = Espresso.onView(withId(R.id.text_input_password_toggle))
+        checkableImageButton.perform(ViewActions.click())
+        checkableImageButton.perform(ViewActions.click())
+        not(isTextDisplayed("1234567890"))
+
+    }
+
+
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt]: delete

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    @Test
+    fun tapEyeToHidePassword() {
+        shouldNotSeePassword()
+        val checkableImageButton = Espresso.onView(withId(R.id.text_input_password_toggle))
+        checkableImageButton.perform(ViewActions.click())
+        checkableImageButton.perform(ViewActions.click())
+        not(isTextDisplayed("1234567890"))
+
+    }
+
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt]: delete

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPhoneNumberFieldTest.kt [hunk 7]
```diff
@@ -96,6 +96,7 @@ class EditTextPhoneNumberFieldTest : MockActivityTest() {
         val phone = "123456"
         val txtInputLayout = (ruleActivity.activity.findViewById<View>(R.id.tlPhone) as? EditTextPhoneField)
         txtInputLayout?.setIsRequired(true)
+        txtInputLayout?.editText
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPhoneNumberFieldTest.kt]: ?

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/eye.kt [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.TypeSafeMatcher
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class eye {
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/eye.kt]: delete class

File: test/src/main/java/co/condorlabs/customcomponents/test/MockActivity.kt [hunk 9]
```diff
@@ -35,6 +35,6 @@ class MockActivity : AppCompatActivity(), ValidableView {
     }
 
     companion object {
-        var layout: Int = R.layout.activity_mock
+        var layout: Int = R.layout.activity_passwordtestfield_test
```
Comment (leidyzulu) [inline test/src/main/java/co/condorlabs/customcomponents/test/MockActivity.kt]: change to activity_mock

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/BaseEditTextFormField.kt [hunk 10]
```diff
@@ -31,11 +30,7 @@ import android.widget.LinearLayout
 import co.condorlabs.customcomponents.R
 import co.condorlabs.customcomponents.formfield.FormField
 import co.condorlabs.customcomponents.formfield.ValidationResult
-import co.condorlabs.customcomponents.helper.DEFAULT_STYLE_ATTR
-import co.condorlabs.customcomponents.helper.DEFAULT_STYLE_RES
-import co.condorlabs.customcomponents.helper.EMPTY
-import co.condorlabs.customcomponents.helper.VALIDATE_EMPTY_ERROR
-import co.condorlabs.customcomponents.helper.VALIDATE_INCORRECT_ERROR
+import co.condorlabs.customcomponents.helper.*
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/BaseEditTextFormField.kt]: import each constant

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextCurrencyFieldTest.kt [hunk 11]
```diff
@@ -34,6 +34,7 @@ class EditTextCurrencyFieldTest : MockActivityTest() {
 
     @Before
     fun setup() {
+
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextCurrencyFieldTest.kt]: delete

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+        isTextDisplayed("1234567890")
+
+    }
+
+    @Test
+    fun tapEyeToHidePassword() {
+        shouldNotSeePassword()
+        val checkableImageButton = Espresso.onView(withId(R.id.text_input_password_toggle))
+        checkableImageButton.perform(ViewActions.click())
+        checkableImageButton.perform(ViewActions.click())
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt]: it's necessary duplicate this line?

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextPasswordField.kt (hunk 1 above)
Comment (ogallon) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextPasswordField.kt]: hint should be a dynamic thing, so do not override his value

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+import co.condorlabs.customcomponents.helper.VALIDATE_EMPTY_ERROR
+import co.condorlabs.customcomponents.test.util.isTextDisplayed
+import org.hamcrest.Matchers.not
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class EditTextPasswordFieldTest : MockActivityTest() {
```
Comment (ogallon) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/EditTextPasswordFieldTest.kt]: missing test, should display password toggle icon

File: test/src/main/res/layout/activity_passwordtestfield_test.xml [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+              android:layout_width="match_parent"
+              android:layout_height="match_parent">
+
+    <co.condorlabs.customcomponents.customedittext.EditTextPasswordField
+            android:layout_width="253dp"
```
Comment (ogallon) [inline test/src/main/res/layout/activity_passwordtestfield_test.xml]: use match parent

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextPasswordField.kt [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    override fun getErrorValidateResult(): ValidationResult {
+        return ValidationResult(false, VALIDATE_EMPTY_ERROR)
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextPasswordField.kt]: this method it's call when appear an error with regex so VALIDATE EMPTY ERROR is not correct

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/BaseEditTextFormField.kt (hunk 10 above)
Comment (otigreros) [reply app/src/main/java/co/condorlabs/customcomponents/customedittext/BaseEditTextFormField.kt]: Eso, lo pone automaticamente A.S. cuando se importan 4 constantes del mismo lugar

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextPasswordField.kt [hunk 16]
```diff
@@ -37,10 +38,5 @@ class EditTextPasswordField(context: Context, attrs: AttributeSet) : BaseEditTex
         mEditText?.id = R.id.etPassword
         mEditText?.transformationMethod = PasswordTransformationMethod.getInstance()
         isPasswordVisibilityToggleEnabled = true
-        mHint = "Password"
-    }
-
-    override fun getErrorValidateResult(): ValidationResult {
-        return ValidationResult(false, VALIDATE_EMPTY_ERROR)
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextPasswordField.kt]: why it was deleted?

File: app/src/main/java/co/condorlabs/customcomponents/helper/Constants.kt [hunk 17]
```diff
@@ -98,3 +98,4 @@ const val COMPOUND_DRAWABLE_POSITION_ARRAY_SIZE = 2
 const val DATE_EDIT_TEXT_RIGHT_COMPOUND_DRAWABLE_POSITION = 0
 const val DRAWABLE_RIGHT_POSITION = 2
 const val COMPOUND_DRAWABLE_TOUCH_OFF_SET = 32
+const val PASSWORD = "Password"
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/helper/Constants.kt]: unused

**Pull request merged**